dio/lib/src/dio_exception.dart

```diff
@@ -151,13 +151,14 @@ class DioException implements Exception {
   factory DioException.connectionError({
     required RequestOptions requestOptions,
     required String reason,
+    Object? error,
```

Comment:
cc @ueman to check if adding the field would affect something.

Reply:
This is safe to do, because it's not a required field